experimental/packages/opentelemetry-sdk-metrics-base/src/state/DeltaMetricProcessor.ts

```diff
@@ -35,31 +35,36 @@ export class DeltaMetricProcessor<T extends Maybe<Accumulation>> {
 
   constructor(private _aggregator: Aggregator<T>) {}
 
-  /** Bind an efficient storage handle for a set of attributes. */
-  private bind(attributes: MetricAttributes) {
-    return this._activeCollectionStorage.getOrDefault(attributes, () => this._aggregator.createAccumulation());
-  }
-
-  record(value: number, attributes: MetricAttributes, _context: Context) {
-    const accumulation = this.bind(attributes);
+  record(value: number, attributes: MetricAttributes, _context: Context, collectionTime: HrTime) {
+    const accumulation = this._activeCollectionStorage.getOrDefault(
+      attributes,
+      () => this._aggregator.createAccumulation(collectionTime)
+    );
     accumulation?.record(value);
   }
 
-  batchCumulate(measurements: AttributeHashMap<number>) {
+  batchCumulate(measurements: AttributeHashMap<number>, collectionTime: HrTime) {
     Array.from(measurements.entries()).forEach(([attributes, value, hashCode]) => {
-      let accumulation = this._aggregator.createAccumulation();
+      const accumulation = this._aggregator.createAccumulation(collectionTime);
       accumulation?.record(value);
+      let delta = accumulation;
       if (this._cumulativeMemoStorage.has(attributes, hashCode)) {
+        // previous must present.
         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
         const previous = this._cumulativeMemoStorage.get(attributes, hashCode)!;
-        accumulation = this._aggregator.diff(previous, accumulation);
+        delta = this._aggregator.diff(previous, accumulation);
       }
 
+      // Save the current record and the delta record.
       this._cumulativeMemoStorage.set(attributes, accumulation, hashCode);
-      this._activeCollectionStorage.set(attributes, accumulation, hashCode);
+      this._activeCollectionStorage.set(attributes, delta, hashCode);
     });
   }
 
+  /**
+   * Returns a collection of delta metrics. Their start time is the when first
+   * time event collected.
+   */
```

Comment:
The start time in the delta metric processor is not the time of the previous collection event. The time of first event reported in delta metric processor may not be as early as the previous collection event:

```
- t0: event
- t1: collection
- t2: event
- t3: event
- t4: collection
```

In this case, the start time reported at t4 for this metric stream is t2 rather than t1.

Reply:
Ah, right. My mistake, sorry. Then this is good. :slightly_smiling_face: